woocommerce/woocommerce-ios
# [Subscriptions in Order Creation 3] Update signup fees by product quantity

Closes: #12564

## Description
This PR updates the product card UI to display signup fees correctly when a subscription item quantity is > 1. Up until now we would show only a single signup fee, but this needs to be multiplied by the item quantity:

![Simulator Screen Recording - iPhone 15 - 2024-04-26 at 10 29 54](https://github.com/woocommerce/woocommerce-ios/assets/3812076/ac7378ba-d076-4051-ac97-ff9f635e4f1d)


## Changes
- Update signup fees when quantity is updated
- Added a _"quantity x price"_ line to signup fees when quantity is > 1
- Extracted duplication when computing price * quantity into a helper function

## Testing instructions
* On a store using WooCommerce Subscriptions (you can use the testing site https://indiemelon.mystagingwebsite.com/)
* Tap `Orders` > `+` > `+ Add Product` > search for a subscription product with signup fees (named as "virtual sub" on my test store) > Add it to the order
* Expand the product card and observe that the signup fee line is updated when quantity changes, as well as the fee summary _(quantity x signup fee)_ appears and is updated when quantity > 1

Question: When we have only one subscription product with signup fees added to the order, I opted to not show the _"price x quantity"_ line unless quantity is > 1, since just adds unnecessary information to an already packed view. For consistency, I think we should also remove it from the product price line if quantity is 1,  so the view gets simpler. WDTY?
<img width=450 src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/e2a5f2f6-336d-4437-95c4-c0035e3af001"/>
